### PR TITLE
Change some types from String to Snowflake

### PR DESCRIPTION
**Pull request by Lukellmann**
- https://github.com/kordlib/kord/commit/d109712a47579a20ea2a8233f1bb1a77c592f28b: `DiscordGuildBan.guildId`
  see:
  https://discord.com/developers/docs/topics/gateway#guild-ban-add-guild-ban-add-event-fields
  https://discord.com/developers/docs/topics/gateway#guild-ban-remove-guild-ban-remove-event-fields
- https://github.com/kordlib/kord/commit/9784b5a095404a1ddd7d9618e384e39548513c7e: `DMCreatRequest.userId`
  see: https://discord.com/developers/docs/resources/user#create-dm
-  https://github.com/kordlib/kord/commit/6dfd86167b0c3e2fa30dee88c86b44d6f9b03f96: `ChannelFollowRequest.webhookChannelId`
  see: https://discord.com/developers/docs/resources/channel#follow-news-channel

### CODE REVIEW

File: common/src/main/kotlin/entity/DiscordGuild.kt [hunk 1]
```diff
@@ -155,8 +155,8 @@ data class DiscordGuild(
     @SerialName("welcome_screen")
     val welcomeScreen: Optional<DiscordWelcomeScreen> = Optional.Missing(),
     @SerialName("nsfw_level")
-    val nsfwLevel: NsfwLevel
-    )
+    val nsfwLevel: NsfwLevel,
```
Comment (DRSchlaubi) [inline common/src/main/kotlin/entity/DiscordGuild.kt:158]: Ouch

File: common/src/main/kotlin/entity/DiscordGuild.kt [hunk 2]
```diff
@@ -176,7 +176,7 @@ class DiscordPartialGuild(
     val permissions: Optional<Permissions> = Optional.Missing(),
     val features: List<GuildFeature>,
     @SerialName("welcome_screen")
-    val welcomeScreen: Optional<DiscordWelcomeScreen> = Optional.Missing()
+    val welcomeScreen: Optional<DiscordWelcomeScreen> = Optional.Missing(),
```
Comment (DRSchlaubi) [inline common/src/main/kotlin/entity/DiscordGuild.kt:179]: Ouch 2

File: common/src/main/kotlin/entity/DiscordGuild.kt [hunk 3]
```diff
@@ -402,7 +402,7 @@ data class DiscordVoiceState(
     val selfStream: OptionalBoolean = OptionalBoolean.Missing,
     val suppress: Boolean,
     @SerialName("request_to_speak_timestamp")
-    val requestToSpeakTimestamp: String?
+    val requestToSpeakTimestamp: String?,
```
Comment (DRSchlaubi) [inline common/src/main/kotlin/entity/DiscordGuild.kt:405]: Ouch 3

File: common/src/main/kotlin/entity/DiscordGuild.kt [hunk 4]
```diff
@@ -613,12 +614,12 @@ data class DiscordWelcomeScreenChannel(
     @SerialName("emoji_id")
     val emojiId: Snowflake?,
     @SerialName("emoji_name")
-    val emojiName: String?
+    val emojiName: String?,
```
Comment (DRSchlaubi) [inline common/src/main/kotlin/entity/DiscordGuild.kt:617]: Ouch whatever comes after 3

File: common/src/main/kotlin/entity/DiscordGuild.kt [hunk 5]
```diff
@@ -613,12 +614,12 @@ data class DiscordWelcomeScreenChannel(
...
 )
 
 @Serializable
 data class DiscordWelcomeScreen(
     val description: String?,
     @SerialName("welcome_channels")
-    val welcomeChannels: List<DiscordWelcomeScreenChannel>
+    val welcomeChannels: List<DiscordWelcomeScreenChannel>,
```
Comment (DRSchlaubi) [inline common/src/main/kotlin/entity/DiscordGuild.kt:624]: Ouch 5

File: core/src/main/kotlin/behavior/UserBehavior.kt [hunk 6]
```diff
@@ -106,7 +106,7 @@ interface UserBehavior : KordEntity, Strategizable {
 fun UserBehavior(
     id: Snowflake,
     kord: Kord,
-    strategy: EntitySupplyStrategy<*> = kord.resources.defaultStrategy
+    strategy: EntitySupplyStrategy<*> = kord.resources.defaultStrategy,
```
Comment (DRSchlaubi) [inline core/src/main/kotlin/behavior/UserBehavior.kt:109]: !

File: core/src/main/kotlin/behavior/channel/NewsChannelBehavior.kt [hunk 7]
```diff
@@ -63,14 +62,14 @@ interface NewsChannelBehavior : ThreadParentChannelBehavior {
...
-        kord.rest.channel.followNewsChannel(id, ChannelFollowRequest(webhookChannelId = target.asString))
+        kord.rest.channel.followNewsChannel(id, ChannelFollowRequest(webhookChannelId = target))
     }
 
 
     suspend fun startPublicThread(
         name: String,
         archiveDuration: ArchiveDuration = ArchiveDuration.Day,
-        reason: String? = null
+        reason: String? = null,
```
Comment (DRSchlaubi) [inline core/src/main/kotlin/behavior/channel/NewsChannelBehavior.kt:72]: !

File: core/src/main/kotlin/behavior/channel/NewsChannelBehavior.kt [hunk 8]
```diff
@@ -79,7 +78,7 @@ interface NewsChannelBehavior : ThreadParentChannelBehavior {
         messageId: Snowflake,
         name: String,
         archiveDuration: ArchiveDuration = ArchiveDuration.Day,
-        reason: String? = null
+        reason: String? = null,
```
Comment (DRSchlaubi) [inline core/src/main/kotlin/behavior/channel/NewsChannelBehavior.kt:81]: !

File: core/src/main/kotlin/behavior/channel/NewsChannelBehavior.kt [hunk 9]
```diff
@@ -101,7 +100,7 @@ fun NewsChannelBehavior(
     guildId: Snowflake,
     id: Snowflake,
     kord: Kord,
-    strategy: EntitySupplyStrategy<*> = kord.resources.defaultStrategy
+    strategy: EntitySupplyStrategy<*> = kord.resources.defaultStrategy,
```
Comment (DRSchlaubi) [inline core/src/main/kotlin/behavior/channel/NewsChannelBehavior.kt:103]: !

File: core/src/main/kotlin/gateway/handler/GuildEventHandler.kt [hunk 10]
```diff
@@ -27,7 +27,7 @@ internal class GuildEventHandler(
     kord: Kord,
     gateway: MasterGateway,
     cache: DataCache,
-    coreFlow: MutableSharedFlow<CoreEvent>
+    coreFlow: MutableSharedFlow<CoreEvent>,
```
Comment (DRSchlaubi) [inline core/src/main/kotlin/gateway/handler/GuildEventHandler.kt:30]: !

File: rest/src/main/kotlin/json/request/ChannelFollowRequest.kt [hunk 11]
```diff
@@ -1,10 +1,11 @@
...
 
+import dev.kord.common.entity.Snowflake
 import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
 
 @Serializable
 class ChannelFollowRequest(
     @SerialName("webhook_channel_id")
-    val webhookChannelId: String
+    val webhookChannelId: Snowflake,
```
Comment (DRSchlaubi) [inline rest/src/main/kotlin/json/request/ChannelFollowRequest.kt:10]: !

File: rest/src/main/kotlin/json/request/UserRequests.kt [hunk 12]
```diff
@@ -1,31 +1,32 @@
...
+import dev.kord.common.entity.Snowflake
 import dev.kord.common.entity.optional.Optional
 import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
 
 @Serializable
 data class DMCreateRequest(
     @SerialName("recipient_id")
-    val userId: String
+    val userId: Snowflake,
```
Comment (DRSchlaubi) [inline rest/src/main/kotlin/json/request/UserRequests.kt:11]: !

File: rest/src/main/kotlin/json/request/UserRequests.kt [hunk 13]
```diff
@@ -1,31 +1,32 @@
...
 )
 
 @Serializable
 data class GroupDMCreateRequest(
     @SerialName("access_tokens")
     val tokens: List<String>,
-    val nick: Map<String, String>
+    val nick: Map<String, String>,
```
Comment (DRSchlaubi) [inline rest/src/main/kotlin/json/request/UserRequests.kt:18]: !

File: rest/src/main/kotlin/json/request/UserRequests.kt [hunk 14]
```diff
@@ -1,31 +1,32 @@
...
 )
 
 @Serializable
 data class CurrentUserModifyRequest(
     val username: Optional<String> = Optional.Missing(),
-    val avatar: Optional<String?> = Optional.Missing()
+    val avatar: Optional<String?> = Optional.Missing(),
```
Comment (DRSchlaubi) [inline rest/src/main/kotlin/json/request/UserRequests.kt:24]: !

File: rest/src/main/kotlin/json/request/UserRequests.kt [hunk 15]
```diff
@@ -1,31 +1,32 @@
...
 )
 
 @Serializable
 data class UserAddDMRequest(
     @SerialName("access_token")
     val token: String,
-    val nick: String
-)
\ No newline at end of file
+    val nick: String,
```
Comment (DRSchlaubi) [inline rest/src/main/kotlin/json/request/UserRequests.kt:31]: !

**Review by DRSchlaubi — CHANGES_REQUESTED** [review]
What's p with all of these trailing `,`

File: rest/src/main/kotlin/json/request/UserRequests.kt (hunk 15 above)
Comment (Lukellmann) [reply rest/src/main/kotlin/json/request/UserRequests.kt:31]: > What's p with all of these trailing `,`

My formatting settings (https://kotlinlang.org/docs/coding-conventions.html#trailing-commas) and me hitting Ctrl+Alt+L all the time, should I leave them out?

**Pull request closed without merging**